app/models/assignment.rb

```diff
@@ -0,0 +1,7 @@
+class Assignment < ApplicationRecord
+  belongs_to :user
+  belongs_to :offspring
+  belongs_to :shift
```

Comment:
You need to add has_many to shifts, has_many to user (he can have many offspring)
And a check that if the assignment belongs to offspring, the user is the parent of the offpsring